### ISSUE TITLE
Incorrect Content-Length header with multibyte characters with HttpContent.Complete

### ISSUE THREAD

**Issue by lilatomic** [dream11/zio-http#176]
Hello,

When using HttpContent.Complete with a multibyte character, the calculation for the Content-Length header does not take into account that this is a multibyte character. This is because of the length is calculated with `String.length()` instead of with `String.getBytes.length` . Illustration:
``` scala
"λ".length() // 1
"λ".getBytes.length // 2
```

With a minimal example:
``` scala
object Main extends zio.App {
	val app = Http.collect{ case Method.GET -> Root => Response.text("λλ") }
	override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] = Server.start(8080, app).exitCode
}
```
we can use `curl -iv --raw localhost:8080` to see that we get `content-length: 2`, an "Excess found" of size 2, and only one "λ"
```
* Excess found in a non pipelined read: excess = 2, size = 2, maxdownload = 2, bytecount = 0
* Connection #0 to host localhost left intact
λ⏎                
```

This is in the latest release [1.0.0.0-RC15](https://github.com/dream11/zio-http/blob/v1.0.0.0-RC15/zio-http/src/main/scala/zhttp/service/HttpMessageCodec.scala#L49) and on [main](https://github.com/dream11/zio-http/blob/main/zio-http/src/main/scala/zhttp/service/EncodeRequest.scala#L22). Let me know if you want me to put together an MR.

**Comment by tusharmath** [comment]
@lilatomic can you try the latest snapshot release also? I think it's fixed there.

**Comment by lilatomic** [comment]
yep, it looks like it's fixed in the snapshot 1.0.0.0-RC15+31-46c879fd-SNAPSHOT . I guess that's due to the change from `HttpContent.Complete(data)` to `HttpData.CompleteData(Chunk.fromArray(data.getBytes(HTTP_CHARSET)))` in `ResponseOps.text`. That's a bit less ergonomic for creating responses which aren't text or json (I was creating some barebones HTML responses when I noticed this). I think it would be nice to have a simple `HttpData.fromString` method for that use case (and internally, possibly):
``` scala
	def fromString(s: String): HttpData[Any, Nothing] = HttpData.CompleteData(Chunk.fromArray(s.getBytes(HTTP_CHARSET)))
```
anyhow, thanks for your help and the cool library!
